.github/actions/app-create/action.yml

```diff
@@ -84,10 +89,12 @@ runs:
         rm -rf "${APP_DIR}"
 
         # Roll out the project
-        printf "%s\n%s\n\n\n\n\n\n\n\n%s\n" "${APP_NAME}" "${APP_DIR}" "${GUI_INPUT}" \
-          | briefcase new ${TEMPLATE} ${TEMPLATE_BRANCH}
+        briefcase new --no-input ${TEMPLATE} ${TEMPLATE_BRANCH} \
+          -Q "formal_name=${APP_NAME}" \
+          -Q "app_name=${APP_DIR}" \
+          -Q "bootstrap=${BOOTSTRAP}"
```

Comment:
ahhh...of course....this won't work for previous versions of Briefcase that backport PRs for templates will be using...

this almost starts to ask the question "should we have versioned branches for CI....?"

Reply:
So, even if I pivot the invocation of `briefcase new` to use stdin injection for previous releases of Briefcase, there's a more fundamental problem.

When we merged #68, that was actually when we broke the ability to run CI for backport PRs for templates since we removed the expectation in `app-build-verify` for templates to contain an app to test. Instead, we started using `--config` to define the template file path to use to create an app on-the-fly. Of course, if CI is using an older release of Briefcase, `--config` is not going to work.

Reply:
Given this, I think it would be particularly onerous at this point for CI to support previous versions of Briefcase; the complexity is already above where we'd like and this would drastically compound that.

So, I think we have two options:
1) Disable CI for template PRs that are not targeting their respective `main` branch
    - Obviously the simplest choice....and maybe the reasonable choice
    - Backport PRs are rare and are usually just version bumps in `pyproject.toml`
2) Create a new branch in the `beeware/.github` repo that exposes the old CI for template repo branches to use
    - This introduces additional complexity and tech debt....but not nearly as much as trying to support older releases of Briefcase in `main`
    - Implementation should be straightforward...although, tedious. Once a new branch off an older commit exists in `beeware/.github`, all of the versioned branches in the template repos would need to be updated to use it
    - Alternatively, we could avoid such updates until any PRs are actually created for one of those branches

Reply:
Completely agreed that excessive complexity in aid of CI on old template branches isn't worth it. 

I'll go a step further on your "backport PRs are rare" statement, and say that a backport PR for anything other than the most recent version is almost unheard of. The only times we've ever backported a patch is when we've noticed something problematic in the current stable version that can be fixed without a full Briefcase release.

That means that once 0.3.17 is in the wild, all the branches we actually care about *will* be able to run CI. It's just this intermediate stage where 0.3.16 doesn't support the extra arguments that is a problem. If that means we disable app-build-verify for non-main branches (or, better still, "non-main branches before 0.3.17"), I'd be fine with that.

With that said, it does make me wonder if the root cause of the problem is that we're encoding too much logic in this .github repo. The complexity here is mostly because the arguments being passed *to* briefcase new/create/build/run/package are embedded in *this* repo, not the invoking repo. Moving the invocation-specific logic to the calling process might involve a *little* duplication; but it should (AFAICT) significantly simplify the .github workflows, and would allow us to pass different arguments in on a CI *branch*.

Reply:
> With that said, it does make me wonder if the root cause of the problem is that we're encoding too much logic in this .github repo. The complexity here is mostly because the arguments being passed _to_ briefcase new/create/build/run/package are embedded in _this_ repo, not the invoking repo. Moving the invocation-specific logic to the calling process might involve a _little_ duplication; but it should (AFAICT) significantly simplify the .github workflows, and would allow us to pass different arguments in on a CI _branch_.

Soo...this would effectively amount to an entire parametrization of the workflow.....such that any logic would be pushed to the calling repo. This would include:
- Source of the Briefcase project template
- Extra parameters to use to run `briefcase new`
- Source of the Briefcase build template
- Extra steps to perform (such as `apt install`) before building the app

This is all possible....probably wouldn't even be that hard (famous last words 😉)....although, I'm not sure the simplification would be particularly significant; doing this would only really remove 2 steps.....it may even require introducing another step.

There's also an opportunity to improve whether Briefcase is installed from a pre-built package or whether the version to install is derived from the nature of the CI event.

I'm willing to experiment with this idea once beeware/briefcase#1549 is merged....but I think I'm not expecting too big of changes. Please let me know, though, if you're thinking of other levels of changes.

Reply:
Hrm - you may be right that this won't actually simplify thing that much.

I should have a chance to do a deep review of this tomorrow morning; I'll keep this sort of thing in mind when I do that review.

Reply:
Oh....also just remembered we'd have to parameterize whether to run `briefcase create` or use an existing project as well because `--config` won't work for older versions of Briefcase; this, of course, goes along with passing the arguments for `briefcase new`.

I guess in a lot of ways, we've fundamentally changed the expectations for this part of CI.